app/controllers/users_controller.rb

```diff
@@ -83,10 +84,10 @@ def persisted_form_attributes
   end
 
   def group_ids
-    current_user.groups.map { |group| { group_id: group.id } }
+    current_user.groups.map { |group| group.id }
```

Comment:
このぐらいなら
```suggestion
    current_user.groups.map(&:id)
```
でOK
一回しか使わずActiveRecord側にキャッシュする必要ないなら
```ruby
current_user.groups.ids
```
でSQLでidだけ取得するようになりActiveRecordのオブジェクトが作られず速い。eager loadingも必要なくなる

Reply:
> 一回しか使わずActiveRecord側にキャッシュする必要ないなら

という事は↓のメソッドを一回使うと`group_ids`メソッドの返り値がキャッシュされてしまうんですね。

```ruby
def group_ids
  current_user.groups.map(&:id)
end
```

キャッシュって意図せず行われるからなかなか意識してなっかたのですが、上でもキャッシュされてるのはびっくりしました。というかキャッシュについてあまり理解してないのでちょっとキャッシュ関連でググってみようと思いました